test/vm_tests/builder/impl_generation_test.dart

```diff
@@ -81,352 +81,528 @@ main() {
-          expect(implGenerator.outputContentsBuffer.toString(), contains('parentType: baz.\$BarComponentFactory'));
-        });
-      });
-
-      group('and includes concrete accessors class for ', () {
-        void testAccessorGeneration(String testName, OverReactSrc ors) {
-          group(testName, () {
-            bool isProps;
-            String className;
-            String descriptorType;
-            setUp(() {
-              generateFromSource(ors.source);
-              isProps = ors.isProps(ors.annotation);
-              className = isProps ? ors.propsClassName : ors.stateClassName;
-              descriptorType = '${isProps ? 'Prop' : 'State'}Descriptor';
-            });
-
-            test('with proper accessors class declaration, retaining type parameters', () {
-              expect(implGenerator.outputContentsBuffer.toString(), contains(
-                  'abstract class _\$${className}AccessorsMixin${ors.typeParamSrc} '
-                      'implements _\$$className${ors.typeParamSrcWithoutBounds} {'));
-            });
-
-            test('with abstract props/state getter', () {
-              expect(implGenerator.outputContentsBuffer.toString(), contains('@override  Map get ${isProps ? 'props' : 'state'};'));
-            });
-
-            test('contains props or state descriptors for all fields', () {
-              expect(implGenerator.outputContentsBuffer.toString(), contains(
-                  '  static const $descriptorType _\$prop__someField___\$$className = const $descriptorType(_\$key__someField___\$$className);\n'
-                  '  static const $descriptorType _\$prop__foo___\$$className = const $descriptorType(_\$key__foo___\$$className);\n'
-                  '  static const $descriptorType _\$prop__bar___\$$className = const $descriptorType(_\$key__bar___\$$className);\n'
-                  '  static const $descriptorType _\$prop__baz___\$$className = const $descriptorType(_\$key__baz___\$$className);\n'));
-            });
+      void testImplGeneration(String groupName, {backwardsCompatible: true}) {
+        group(groupName, () {
+          test('stateful components', () {
+            generateFromSource(OverReactSrc.state(backwardsCompatible: backwardsCompatible).source);
+          });
 
-            test('contains string keys', () {
-              expect(implGenerator.outputContentsBuffer.toString(), contains(
-                  '  static const String _\$key__someField___\$$className = \'$className.someField\';\n'
-                  '  static const String _\$key__foo___\$$className = \'$className.foo\';\n'
-                  '  static const String _\$key__bar___\$$className = \'$className.bar\';\n'
-                  '  static const String _\$key__baz___\$$className = \'$className.baz\';\n'));
-            });
+          test('component', () {
+            generateFromSource(OverReactSrc.props(backwardsCompatible: backwardsCompatible).source);
```

Comment:
This isn't actually verifying anything is it?

Reply:
Actually, in the `tearDown` for the containing group, there's two verifications: `verifyNoErrorLogs()`, and `verifyImplGenerationIsValid()`.